shell/platform/embedder/embedder.h

```diff
@@ -369,6 +375,14 @@ typedef struct {
   uint32_t name;
   /// The texture format (example GL_RGBA8).
   uint32_t format;
+  /// The pixel data buffer.
+  const uint8_t* buffer;
+  /// The size of pixel buffer.
+  size_t buffer_size;
+  /// Callback invoked that the gpu surface texture start binding.
+  BoolCallback bind_callback;
+  /// The type of the texture.
+  FlutterGLImpellerTextureType impeller_texture_type;
```

Comment:
How about using bool rather than enum type to minimize additional code?
It can be `is_pixel_buffer_texture` or `is_gpu_surface_texture`.